src/language-markdown/parser-markdown.js

```diff
@@ -68,13 +68,20 @@ function htmlToJsx() {
       }
 
       return nodes.reduce((newNodes, { sourceSpan: position, type }) => {
-        const value = node.value
-          .slice(position.start.offset, position.end.offset)
-          .trim();
-
-        if (value) {
+        const isEl = type === "element";
+        let value = node.value.slice(
+          position.start.offset,
+          position.end.offset
+        );
+
+        if (value.trim()) {
+          const isPrevEl = newNodes[newNodes.length - 1];
+          value = isEl ? value : value.replace(/\s+/g, " "); // alike text in jsx
```

Comment:
Whitespaces should be processed by `splitText` in `utils.js` so you shouldn't need to process them here. And also, the newline processing should be controlled by the `--prose-wrap` option.